### PR TITLE
Resolve resources in controller

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add implementation to resolve resources in controller

**Related issue(s)**
https://github.com/kyma-project/serverless-manager/issues/248

### CODE REVIEW

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go [hunk 1]
```diff
@@ -41,8 +41,6 @@ type DefaultingConfig struct {
 }
 
 func (fn *Function) Default(config *DefaultingConfig) {
-	fn.Spec.defaultFunctionResources(config, fn)
```
Comment (dbadura) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go:44]: I will remove it in next PR to not make this PR very big

**Review by dbadura — COMMENTED** [review]
Add runtime presets

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go (hunk 1 above)
Comment (dbadura) [reply components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go:44]: Here is a draft: https://github.com/kyma-project/kyma/pull/18164

**Review by pPrecel — APPROVED** [review]

File: components/function-controller/README.md [hunk 2]
```diff
@@ -16,7 +16,11 @@ To develop the Function Controller, you need:
 - [controller-gen](https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.6.2) (v0.6.2)
 - <!-- markdown-link-check-disable-line -->[kustomize](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.5.7) (v4.5.7)
 
-To develop the component, use the formulae declared in the [generic](/common/makefiles/generic-make-go.mk) and [component-specific](./Makefile) Makefiles. To run tests without the Makefile logic, use the `go test ./...` command.
+To develop the component, use the formulae declared in the [Makefile](./Makefile).
+
+To run manager and setup envs correctly,  source [controller.env](./hack/controller.env).
```
Comment (grego952) [inline components/function-controller/README.md:21]:
```suggestion
To run the manager and set up envs correctly, source [controller.env](./hack/controller.env). 
```

File: components/function-controller/README.md [hunk 3]
```diff
@@ -16,7 +16,11 @@ To develop the Function Controller, you need:
 - [controller-gen](https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.6.2) (v0.6.2)
 - <!-- markdown-link-check-disable-line -->[kustomize](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.5.7) (v4.5.7)
 
-To develop the component, use the formulae declared in the [generic](/common/makefiles/generic-make-go.mk) and [component-specific](./Makefile) Makefiles. To run tests without the Makefile logic, use the `go test ./...` command.
+To develop the component, use the formulae declared in the [Makefile](./Makefile).
+
+To run manager and setup envs correctly,  source [controller.env](./hack/controller.env). 
+You can customize configuration by editing files in [hack](./hack) dir.
```
Comment (grego952) [inline components/function-controller/README.md:22]:
```suggestion
You can customize the configuration by editing files in [hack](./hack) dir. 
```

**Review by grego952 — APPROVED** [review]

**Review by pPrecel — APPROVED** [review]

**Pull request merged**